work_test.go

```diff
@@ -238,21 +239,19 @@ func TestLockJobAdvisoryRace(t *testing.T) {
 	getBackendID := func(conn *pgx.Conn) int32 {
 		var backendID int32
 		err := conn.QueryRow(`
-			SELECT backendid
-			FROM pg_stat_get_backend_idset() psgb(backendid)
-			WHERE pg_stat_get_backend_pid(psgb.backendid) = pg_backend_pid()
+			SELECT pg_backend_pid()
 		`).Scan(&backendID)
 		if err != nil {
 			panic(err)
 		}
 		return backendID
 	}
-	waitUntilBackendIsWaiting := func (backendID int32, name string) {
+	waitUntilBackendIsWaiting := func(backendID int32, name string) {
 		conn := newConn()
 		i := 0
 		for {
 			var waiting bool
-			err := conn.QueryRow(`SELECT pg_stat_get_backend_waiting($1)`, backendID).Scan(&waiting)
```

Comment:
@bgentry In 9.5 `pg_stat_get_backend_waiting` was a helper function that fetches value from `pg_stat_activity.waiting` column. In 9.6 the column was removed and was replaced by wait_event, which is more explanatory than a bool column; see (https://www.postgresql.org/docs/9.6/static/monitoring-stats.html#WAIT-EVENT-TABLE) for all possible values of `wait_event`. `wait_event` is null if the process does not hold any lock.